### PR TITLE
vz: drop dependency on qemu-img

### PR DESCRIPTION
**Pull request by AkihiroSuda**
It is still recommended to install qemu, as vz driver is still experimental

Fix https://github.com/lima-vm/lima/issues/1507

### CODE REVIEW

File: pkg/nativeimgutil/nativeimgutil.go [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+		rN, rErr := r.Read(buf)
+		if rErr != nil {
+			eof = errors.Is(rErr, io.EOF)
+			if !eof {
+				return n, fmt.Errorf("failed to read: %w", rErr)
+			}
+		}
+		// TODO: qcow2reader should have a method to notify whether buf is zero
+		if bytes.Equal(buf, zeroBuf) {
+			if _, sErr := w.Seek(int64(rN), os.SEEK_CUR); sErr != nil {
```
Comment (balajiv113) [inline pkg/nativeimgutil/nativeimgutil.go:130]: os.SEEK_CUR is deprecated. 

We could use the io.SeekCurrent

File: pkg/nativeimgutil/nativeimgutil.go [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+			if wN != rN {
+				return n, fmt.Errorf("read %d, but wrote %d bytes", rN, wN)
+			}
+		}
+	}
+	return n, nil
+}
+
+func MakeSparse(f *os.File, n int64) error {
+	if _, err := f.Seek(n, os.SEEK_SET); err != nil {
```
Comment (balajiv113) [inline pkg/nativeimgutil/nativeimgutil.go:152]: os.SEEK_SET is deprecated.

We could use the io.SeekStart

File: pkg/nativeimgutil/nativeimgutil.go [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+		return fmt.Errorf("failed to call copySparse(), bufSize=%d, copied=%d: %w", bufSize, copied, err)
+	}
+
+	// Resize
+	if size != nil {
+		if err = MakeSparse(destTmpF, *size); err != nil {
+			return err
+		}
+	}
+	if err = destTmpF.Close(); err != nil {
```
Comment (balajiv113) [inline pkg/nativeimgutil/nativeimgutil.go:81]: In case of vm starting stopped during conversion of disk, the temp files are not cleared

Comment (AkihiroSuda) [reply pkg/nativeimgutil/nativeimgutil.go:81]: The line 65 has `os.RemoveAll(destTmp)`

File: pkg/nativeimgutil/nativeimgutil.go (hunk 1 above)
Comment (AkihiroSuda) [reply pkg/nativeimgutil/nativeimgutil.go]: Fixed

File: pkg/nativeimgutil/nativeimgutil.go (hunk 2 above)
Comment (AkihiroSuda) [reply pkg/nativeimgutil/nativeimgutil.go]: Fixed

File: pkg/nativeimgutil/nativeimgutil.go (hunk 3 above)
Comment (balajiv113) [reply pkg/nativeimgutil/nativeimgutil.go:81]: Ya i noticed but on testing this usecase, i could see stale temp diffdisk files.

Steps i did,
- limactl  start template://experimental/vz
- ctrl+c when we get that converting log
- Noticed the diffdisk-*.temp in directory

Comment (AkihiroSuda) [reply pkg/nativeimgutil/nativeimgutil.go:81]: `limactl stop` now remove `*.tmp`

**Review by balajiv113 — APPROVED** [review]
LGTM :)

One thing i noticed is, the conversion takes more time than before. But we can improve on that incrementally as well if possible.

**Pull request merged**